src/PyDynamic/misc/tools.py

```diff
@@ -160,6 +160,92 @@ def trimOrPad(
         return array[0:length]
 
 
+def trimOrPad_ND(
+    array: Union[List, np.ndarray],
+    length: Union[int, tuple],
+    mode: Optional[str] = "constant",
+    real_imag_type=False,
+):
+    """Trim or pad (with zeros) a vector/array to the desired length(s)
+
+    Either trim or zero-pad an array to achieve the required `length`. Both actions
+    are applied to the end of (each axis of) the array.
+
+    Parameters
+    ----------
+    array : list, ND np.ndarray
+        original data
+    length : int, tuple of int
+        length of output
+    mode : str, optional
+        handed over to np.pad, default "constant"
+    real_imag_type : bool
+        if array is to be interpreted as PyDynamic real-imag-type
+        only works for 1D and square-2D arrays
+
+    Returns
+    -------
+    array_modified : np.ndarray of shape (length,)
+        An either trimmed or zero-padded array
+    """
+
+    # force numpy array
+    array = np.array(array)
+
+    # split and reassemble if vector/covariance is in real-imag representation
+    # (only works for 1D and 2D-square arrays of even length)
+    if real_imag_type:
+        N = array.shape[0] // 2
+        kwargs = {
+            "length": length,
+            "mode": "constant",
+            "real_imag_type": False,
+        }
+
+        if len(array.shape) == 1:
+            REAL = trimOrPad_ND(array[:N], **kwargs)
+            IMAG = trimOrPad_ND(array[N:], **kwargs)
+            return np.r_[REAL, IMAG]
+
+        elif is_2d_square_matrix(array):
+            RR = trimOrPad_ND(array[:N, :N], **kwargs)
+            RI = trimOrPad_ND(array[:N, N:], **kwargs)
+            IR = trimOrPad_ND(array[N:, :N], **kwargs)
+            II = trimOrPad_ND(array[N:, N:], **kwargs)
+
+            return np.block([[RR, RI], [IR, II]])
+
+        else:
+            raise ValueError(
+                f"Array of shape {array.shape} cannot be interpreted as real/imag representation."
+            )
```

Comment:
We should insert a line break, not to exceed the maximum line length of 88 chars.

```suggestion
            raise ValueError(
                f"Array of shape {array.shape} "
                "cannot be interpreted as real/imag representation."
            )
```

Reply:
In fact I would like to propose to include some kind of suggestion how to fix the error like "first dimension is expected to be even" or appropriate.

Reply:
Resolved by  1d4acfc.